examples/advanced/geane/macros/runMC.C

```diff
@@ -0,0 +1,149 @@
+
+  Double_t theta    = 10.;
+
+  TString outDir = "./";
+
+  // Output file name
+  TString outFile = Form("%s/geane.mc.root",
+                         outDir.Data());
+
+  // Parameter file name
+  TString parFile = Form("%s/geane.par.root",
+			 outDir.Data());
+
+  TString geoFile = "geofile_" + mcEngine + "_full.root";
+
+  // In general, the following parts need not be touched
+  // ========================================================================
+
+  // ----    Debug option   -------------------------------------------------
+  gDebug = 0;
+  // ------------------------------------------------------------------------
+
+  // -----   Timer   --------------------------------------------------------
+  TStopwatch timer;
+  timer.Start();
+  // ------------------------------------------------------------------------
+
+  // -----   Create simulation run   ----------------------------------------
+  FairRunSim* run = new FairRunSim();
+  run->SetName(mcEngine);              // Transport engine
+  //  run->SetSimulationConfig(new FairVMCConfig());
+  run->SetIsMT(isMT);                  // Multi-threading mode (Geant4 only)
+  run->SetSink(new FairRootFileSink(outFile));          // Output file
+  FairRuntimeDb* rtdb = run->GetRuntimeDb();
+  // ------------------------------------------------------------------------
+
+  // -----   Create media   -------------------------------------------------
+  run->SetMaterials("media.geo");       // Materials
+  // ------------------------------------------------------------------------
+
+  // -----   Create geometry   ----------------------------------------------
+
+  FairModule* cave= new FairCave("CAVE");
+  cave->SetGeometryFileName("cave_vacuum.geo");
+  run->AddModule(cave);
+
+  FairTutGeaneDet*  det = new FairTutGeaneDet("TutGeaneDetector", kTRUE);
+  det->SetGeometryFileName("pixel.geo");
+//  det->SetMisalignDetector(kTRUE);
+  run->AddModule(det);
+  // ------------------------------------------------------------------------
+
+  // -----   Create PrimaryGenerator   --------------------------------------
+  FairPrimaryGenerator* primGen = new FairPrimaryGenerator();
+  FairBoxGenerator* boxGen = new FairBoxGenerator(partPdgC[chosenPart], 1);
+
+  boxGen->SetThetaRange (   theta,   theta+0.01);
+  boxGen->SetPRange     (momentum,momentum+0.01);
+  boxGen->SetPhiRange   (0.,360.);
+  boxGen->SetDebug(kTRUE);
+
+  primGen->AddGenerator(boxGen);
+
+  FairConstField *fMagField=new FairConstField();
+  fMagField->SetField(0.,0.,20.); // values are in kG
+  fMagField->SetFieldRegion(-150, 150, -150, 150, -250, 250);// values are in cm (xmin,xmax,ymin,ymax,zmin,zmax)
+  run->SetField(fMagField);
+
+  run->SetGenerator(primGen);
+  // ------------------------------------------------------------------------
+
+  // -----   Initialize simulation run   ------------------------------------
+  UInt_t randomSeed = 123456;
+  TRandom3 random(randomSeed);
+  gRandom = &random;
```

Comment:
Assigning to `gRandom` will leak its previously held object because `gRandom` is an owning pointer... (https://root.cern.ch/doc/master/TRandom3_8cxx_source.html#l00064)

So here it shall be:
```cpp
TRandom* oldGRandom = gRandom;
UInt_t randomSeed = 123456;
gRandom = new TRandom3(randomSeed);
if(oldGRandom) delete oldGRandom;
```

On a second note, I would have also asked if it is wise to hard-code the seed like this in an example without any comment, but looks like `gRandom` is hard-coded with a seed value of `4357` anyways already (https://root.cern.ch/doc/master/classTRandom3.html#aa0f90fdd325edead6b80afcb05099348).

Reply:
Maybe we should just do

```cpp
UInt_t randomSeed = 123456;
gRandom->SetSeed(randomSeed);
```
?